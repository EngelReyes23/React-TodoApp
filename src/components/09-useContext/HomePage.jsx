import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const HomePage = () => {
	// useContext
	const { user } = useContext(UserContext);
	console.log(user);

	return (
		<>
			<h1>HomePage</h1>
			<hr />

			<pre>{JSON.stringify(user, null, 5)}</pre>
		</>
	);
};

export default HomePage;
