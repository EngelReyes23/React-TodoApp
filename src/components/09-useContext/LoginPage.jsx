import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const LoginPage = () => {
	// Obtener la referencia a UserContext
	const { setUser } = useContext(UserContext);

	const user = {
		id: new Date().getTime(),
		name: "Engel",
		email: "correo@correo.com",
	};

	return (
		<>
			<h1>Login Page</h1>
			<hr />
			<button className="btn btn-primary" onClick={() => setUser(user)}>
				Login
			</button>
		</>
	);
};

export default LoginPage;
