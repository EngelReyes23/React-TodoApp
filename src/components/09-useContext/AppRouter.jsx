import React from "react";
import {
	BrowserRouter as Router,
	Switch,
	Route,
	Redirect,
	// Link
} from "react-router-dom";
import NavBar from "./NavBar";
import AboutPage from "./AboutPage";
import HomePage from "./HomePage";
import loginPage from "./LoginPage";

const AppRouter = () => {
	return (
		<Router>
			<div>
				<NavBar />
				<div className="container">
					<Switch>
						<Route exact path="/" component={HomePage} />
						<Route exact path="/login" component={loginPage} />
						<Route exact path="/about" component={AboutPage} />
						<Redirect to="/" />
					</Switch>
				</div>
			</div>
		</Router>
	);
};

export default AppRouter;
