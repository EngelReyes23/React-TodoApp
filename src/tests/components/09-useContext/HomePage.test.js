import { mount } from "enzyme";
import HomePage from "../../../components/09-useContext/HomePage";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en <HomePage />", () => {
	const user = {
		name: "Engel",
		email: "correo@correo.com",
	};

	// 1er test
	test("mostrar el componente correctamente", () => {
		const wrapper = mount(
			<UserContext.Provider value={{ user }}>
				<HomePage />
			</UserContext.Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
