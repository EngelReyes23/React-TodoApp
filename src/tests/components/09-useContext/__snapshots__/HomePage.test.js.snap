// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomePage /> mostrar el componente correctamente 1`] = `
Array [
  <h1>
    HomePage
  </h1>,
  <hr />,
  <pre>
    {
     "name": "Engel",
     "email": "correo@correo.com"
}
  </pre>,
]
`;
